http_proxy.go

```diff
@@ -86,6 +86,7 @@ type HTTPProxyConfig struct {
 	CloseAfterReply        bool
 	DenyDomains            []*regexp.Regexp
 	DenyDomainsExclude     []*regexp.Regexp
+	LogHTTPRequestIDHeader string
```

Comment:
I do not like the practice of adding to the end of the list in random moments it does not make sense.

I think here it should also be named RequestIDHeader and live above RequestModifiers. WDYT?

Reply:
Will do it in #414